pkgs/development/web/bun/default.nix

```diff
@@ -0,0 +1,21 @@
+{ stdenvNoCC, callPackage, lib }:
+let
+  version = "0.1.1";
+  pname = "bun";
+  meta = with lib; {
+    homepage = "https://bun.sh";
+    changelog = "https://github.com/Jarred-Sumner/bun/releases/tag/bun-v${version}";
+    description = "Incredibly fast JavaScript runtime, bundler, transpiler and package manager – all in one";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    longDescription = ''
+      All in one fast & easy-to-use tool. Instead of 1,000 node_modules for development, you only need bun.
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ DAlperin jk ];
+    platforms = [ "x86_64-linux" "x86_64-darwin" "aarch64-darwin" ];
+  };
+in
+assert lib.assertMsg (stdenvNoCC.isDarwin || stdenvNoCC.isLinux) "Only supported on darwin and linux";
+if stdenvNoCC.isDarwin
+then callPackage ./darwin.nix { inherit pname version meta; }
+else callPackage ./linux.nix { inherit pname version meta; }
```

Comment:
Why such an ugliness? The expressions are too similar to be treated in separate files.

Reply:
Fixed